clar2wasm/src/words/equal.rs

```diff
@@ -0,0 +1,89 @@
+use walrus::ir::BinaryOp;
+
+use super::Word;
+
+#[derive(Debug)]
+pub struct IsEq;
+
+impl Word for IsEq {
+    fn name(&self) -> ClarityName {
+        "is-eq".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        // Traverse the first operand pushing it onto the stack
+        let first_op = args.get_expr(0)?;
+        generator.traverse_expr(builder, first_op)?;
+
+        // Save the first_op to a local to be further used.
+        // This allows to use the first_op value without
+        // traversing again the expression.
+        let ty = generator
+            .get_expr_type(first_op)
+            .expect("is-eq value expression must be typed")
+            .clone();
+        let wasm_types = clar2wasm_ty(&ty);
+        let mut val_locals = Vec::with_capacity(wasm_types.len());
+        for local_ty in wasm_types.iter().rev() {
+            let local = generator.module.locals.add(*local_ty);
+            val_locals.push(local);
+            builder.local_set(local);
+        }
+        val_locals.reverse();
+
+        if args.len() == 1 {
+            builder.i32_const(1);
+            return Ok(());
+        } else if args.len() >= 2 {
+            // Get first operand from the local and put it onto stack.
+            for val in &val_locals {
+                builder.local_get(*val);
+            }
+
+            // Traverse the second operand pushing it onto the stack.
+            let sec_op = args.get_expr(1)?;
+            generator.traverse_expr(builder, sec_op)?;
+        }
+
+        // Equals expression needs to handle different types.
+        let type_suffix = match ty {
+            // is-eq-int function can be reused to both int and uint types.
+            TypeSignature::IntType | TypeSignature::UIntType => "int",
+            _ => {
+                return Err(GeneratorError::NotImplemented);
+            }
+        };
+        let func = generator.func_by_name(&format!("is-eq-{}", type_suffix));
+
+        // Call the function with the operands on stack.
+        builder.call(func);
+
+        // Loop through remainder operands, if the case.
+        // First operand will be reused from a local var.
+        for operand in args.iter().skip(2) {
+            // Get first operand from the local and put it onto stack.
+            for val in &val_locals {
+                builder.local_get(*val);
+            }
+
+            // Traverse the next operand and put in onto stack.
+            generator.traverse_expr(builder, operand)?;
+
+            // Call the function with the operands on the stack.
+            builder.call(func);
+
+            // Do an "and" operation with the result from the previous function call.
+            builder.binop(BinaryOp::I32And);
+        }
```

Comment:
How about simplifying this to just loop over `args.iter().skip(1)`? So it would be something like:

```rust
        // Equals expression needs to handle different types.
         let type_suffix = match ty {
             // is-eq-int function can be reused to both int and uint types.
             TypeSignature::IntType | TypeSignature::UIntType => "int",
             _ => {
                 return Err(GeneratorError::NotImplemented);
             }
         };
         let func = generator.func_by_name(&format!("is-eq-{}", type_suffix));

         // Initialize the result to true, in case there is only 1 arg
         builder.i32_const(1);

         // Loop through remainder operands, if the case.
         // First operand will be reused from a local var.
         for operand in args.iter().skip(1) {
             // Get first operand from the local and put it onto stack.
             for val in &val_locals {
                 builder.local_get(*val);
             }

             // Traverse the next operand and put in onto stack.
             generator.traverse_expr(builder, operand)?;

             // Call the function with the operands on the stack.
             builder.call(func);

             // Do an "and" operation with the result from the previous function call.
             builder.binop(BinaryOp::I32And);
         }
```

Reply:
Done, @obycode ... I was about to change to that! 👍